plasmapy/physics/distribution.py

```diff
@@ -58,35 +63,460 @@ def Maxwellian_1D(v: units.m/units.s,
+     Maxwellian distribution function.
+
+    Parameters
+    ----------
+    v: Quantity
+        The speed in units convertible to m/s
+
+    T: Quantity
+        The temperature, preferably in Kelvin
+
+    particle: string, optional
+        Representation of the particle species(e.g., 'p' for protons, 'D+'
+        for deuterium, or 'He-4 +1' for $He_4^{+1}$ : singly ionized helium-4),
+        which defaults to electrons.
+    
+    V_drift: Quantity
+        The drift speed in units convertible to m/s
+        
+    vTh: Quantity, optional
+        Thermal velocity (most probable) in m/s. This is used for
+        optimization purposes to avoid re-calculating vTh, for example
+        when integrating over velocity-space.
+        
+    units: string, optional
+        Selects whether to run function with units and unit checks (when
+        equal to "units") or to run as unitless (when equal to "unitless").
+        The unitless version is substantially faster for intensive
+        computations.
+
+    Returns
+    -------
+    f : Quantity
+        probability in speed^-1, normalized so that:
+        $\int_{0}^{\infty} f(v) dv = 1}
+
+    Raises
+    ------
+    TypeError
+        The parameter arguments are not Quantities and
+        cannot be converted into Quantities.
+
+    UnitConversionError
+        If the parameters is not in appropriate units.
+
+    ValueError
+        If the temperature is negative, or the particle mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    In one dimension, the Maxwellian speed distribution function describing
+    the distribution of particles with speed v in a plasma with temperature T
+    is given by:
+
+    .. math::
+    f(v) = 4 \pi v^2 (\pi * v_Th^2)^{-3/2} \exp(-(v - V_{drift})^2 / v_Th^2)
+    where v_Th = \sqrt(2 k_B T / m) is the thermal speed
+
+
+    Example
+    -------
+    >>> from plasmapy.physics import Maxwellian_speed_1D
+    >>> from astropy import units as u
+    >>> v=1*u.m/u.s
+    >>> Maxwellian_speed_1D(v=v, T= 30000*u.K, particle='e',V_drift=0*u.m/u.s)
+    <Quantity 2.602357544747327e-18 s / m>
+    
     """
+    if units == "units":
+        # unit checks and conversions
+        # checking velocity units
+        v = v.to(u.m/u.s)
+        # catching case where drift velocity has default value, and
+        # needs to be assigned units
+        if V_drift == 0:
+            if not isinstance(V_drift, astropy.units.quantity.Quantity):
+                V_drift = V_drift * u.m/u.s
+        # checking drift velocity units
+        V_drift = V_drift.to(u.m/u.s)
+        # convert temperature to Kelvins
+        T = T.to(u.K, equivalencies=u.temperature_energy())
+        if vTh == -1:
+            # get thermal velocity and thermal velocity squared
+            vTh = (thermal_speed(T, particle=particle, method="most_probable"))
+        elif vTh != -1:
+            # check units of thermal velocity
+            vTh = vTh.to(u.m/u.s)
```

Comment:
If the velocity variable can be used as a velocity (as the tests are passing later), is this check (elif block from line 338) necessary? Not 100% sure.

Reply:
Do you mean that if vTh gets added or multiplied to some other value that has units then astropy will throw an error? I was just trying to be explicit, but that might be a better way to go, less clutter.

Reply:
Well, if it's convertible to `u.m/u.s`, it's not going to throw any errors later even if you don't do the explicit check here. But explicit is neat, I was just asking out of curiosity. Let's keep it this way :)